### PR TITLE
Added Team Production Challenge

### PR DESCRIPTION
**Pull request by Cooldude2606**
Converted team production challenge into the mini game format. Each game consists of a single round of team production. Central team selection area and leader board have been removed.

### CODE REVIEW

**Comment by tovernaar123** [comment]
why does this also have the speedrun.lua files

File: modules/mini-games/team_production/map_scripts.lua [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        local position = {entity.position[1] + offset_x, entity.position[2] + offset_y}
+        if entity.amount then
+          surface.create_entity{name = name, position = position, amount = entity.amount}
+        elseif name == "stack-inserter" then
+          local v = surface.create_entity{name = name, position = position, force = force, direction = entity.direction}
+          v.destructible = false
+          v.minable = false
+          v.rotatable = false
+        elseif name == "red-chest" then
+          local v = surface.create_entity({force = force, name = name, position = position})
```
Comment (tovernaar123) [inline modules/mini-games/team_production/map_scripts.lua:50]: v is not the best variable name 😛 i would make it ent for entity

File: modules/mini-games/team_production/map_scripts.lua [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        count = count + 1
+      end
+    end
+  end
+
+  script_data.surface.set_tiles(map_tiles, true)
+  script_data.surface.regenerate_decorative(nil, chunks)
+end
+
+function Public.recreate_entities(entities, offset_x, offset_y, force, duration, script_data)
```
Comment (tovernaar123) [inline modules/mini-games/team_production/map_scripts.lua:34]: the func has alot lf the same code what i would do is check if the ent is an stack inserter if its not create the ent without direction if it is create it with out. also you can set all other things once and not in all these if statements and use the ifs only for adding them into the table.

File: modules/mini-games/team_production/map_scripts.lua (hunk 1 above)
Comment (Cooldude2606) [reply modules/mini-games/team_production/map_scripts.lua:50]: Not my code so I didnt change it.

File: modules/mini-games/team_production/map_scripts.lua (hunk 2 above)
Comment (Cooldude2606) [reply modules/mini-games/team_production/map_scripts.lua:34]: Not my code so I didnt change it. If it ain't broke, don't fix it.

File: modules/mini-games/team_production/map_scripts.lua (hunk 1 above)
Comment (tovernaar123) [reply modules/mini-games/team_production/map_scripts.lua:50]: fine with me

**Comment by Cooldude2606** [comment]
It has the speedrun.lua files because this branch requires some of the changes that were made to core on the speedrun pr

**Pull request merged**